### ISSUE TITLE
Question About the Embeddings for Chinese Char-Level Dataset?

### ISSUE THREAD

**Issue by aslicedbread** [zalandoresearch/flair#568]
```
from flair.data import TaggedCorpus
from flair.data_fetcher import NLPTaskDataFetcher
import os

columns = {0: 'text', 1: 'ner'}
data_folder = data_path
corpus: TaggedCorpus = NLPTaskDataFetcher.load_column_corpus(data_folder, columns,
                                                              train_file='train',
                                                              test_file='test',
                                                              dev_file='dev')
from flair.embeddings import CharacterEmbeddings
embedding = CharacterEmbeddings()
tag_type = 'ner'
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
print(tag_dictionary.idx2item)
from flair.models import SequenceTagger
tagger: SequenceTagger = SequenceTagger(hidden_size=300,
                                        embeddings=embedding,
                                        tag_dictionary=tag_dictionary,
                                        tag_type=tag_type,
                                        use_crf=True)
from flair.trainers import ModelTrainer
trainer: ModelTrainer = ModelTrainer(tagger, corpus)
trainer.train(train_path,
              learning_rate=0.0001,
              mini_batch_size=32,
              max_epochs=150)

```

I'm a bit confused about the train log during the training about the f1-score & acc & loss as follows:

> 2019-02-25 20:51:34,129 ----------------------------------------------------------------------------------------------------
> 2019-02-25 20:51:34,129 Evaluation method: MICRO_F1_SCORE
> 2019-02-25 20:51:34,130 ----------------------------------------------------------------------------------------------------
> 2019-02-25 20:51:45,006 epoch 1 - iter 0/43 - loss 205.90689087
> 2019-02-25 20:52:23,663 epoch 1 - iter 4/43 - loss 187.82157898
> 2019-02-25 20:53:05,964 epoch 1 - iter 8/43 - loss 187.34028795
> 2019-02-25 20:53:48,759 epoch 1 - iter 12/43 - loss 189.10505089
> 2019-02-25 20:54:33,142 epoch 1 - iter 16/43 - loss 193.72891415
> 2019-02-25 20:55:15,488 epoch 1 - iter 20/43 - loss 194.31980315
> 2019-02-25 20:55:58,236 epoch 1 - iter 24/43 - loss 194.88608887
> 2019-02-25 20:56:38,799 epoch 1 - iter 28/43 - loss 193.13958109
> 2019-02-25 20:57:19,201 epoch 1 - iter 32/43 - loss 191.70639177
> 2019-02-25 20:58:04,476 epoch 1 - iter 36/43 - loss 194.43793921
> 2019-02-25 20:58:43,426 epoch 1 - iter 40/43 - loss 192.13849621
> 2019-02-25 20:58:58,606 ----------------------------------------------------------------------------------------------------
> 2019-02-25 20:58:58,606 EPOCH 1 done: loss 193.1285 - lr 0.0001 - bad epochs 0
> 2019-02-25 20:59:35,938 DEV  : loss 189.05775452 - f-score 0.0002 - acc 0.0001

Is there any problem with my code or does the Character Embeddings support Chinese?
The above situation is the same as my use of Bert-base-Chinese.
```
from flair.embeddings import BertEmbeddings
# init embedding
embedding = BertEmbeddings('bert-base-chinese')
```
Hope to get your reply, THANKS.

**Comment by alanakbik** [comment]
Hello @aslicedbread - the `CharacterEmbeddings` class currently does not support Chinese out of the box, since the default setting is to load a latin alphabet. 

You can adapt the class to Chinese by creating a `Dictionary` object for Chinese and saving it somewhere. You find find more information on how to do this [here](https://github.com/zalandoresearch/flair/issues/179). 

Then, we you instantiate the `CharacterEmbeddings`, you pass to this Dictionary:

```python
embeddings = CharacterEmbeddings(path_to_char_dict='path/to/serialized/dictionary')
```

We haven't tested BERT for Chinese ourselves, but this should work. So it is strange that it is not giving good results.

I would also recommend trying out the BytePairEmbeddings for Chinese: 

```python
embeddings = BytePairEmbeddings('zh')
```

Or best do a `StackedEmbedding` that combines all three :)

**Comment by alanakbik** [comment]
@aslicedbread has this worked for you? I'm closing the issue, but feel free to reopen if you have more questions!

**Comment by aslicedbread** [comment]
Thanks a lot. One more question, need we declare the tagging scheme(BIO or BIOES) which the dataset use in the code?

**Comment by aslicedbread** [comment]
> @aslicedbread has this worked for you? I'm closing the issue, but feel free to reopen if you have more questions!

Thanks a lot. One more question, need we declare the tagging scheme(BIO or BIOES) which the dataset use in the code?

**Comment by alanakbik** [comment]
By default, it expects files to be formatted with a BIO tagging scheme and they get changed into BIOES in the data loader. You can turn the conversion off in the data loader but we generally recommend BIOES.
